File: http/http-server-annotation-processor/src/main/java/ru/tinkoff/kora/http/server/annotation/processor/HttpProcessorException.java
package ru.tinkoff.kora.http.server.annotation.processor;

import javax.annotation.processing.ProcessingEnvironment;
import javax.lang.model.element.Element;
import javax.tools.Diagnostic;

public class HttpProcessorException extends RuntimeException {
    private final Element element;

    public HttpProcessorException(String message, Element element) {
        super(message, null);
        this.element = element;
    }

    public void printError(ProcessingEnvironment processingEnvironment) {
        processingEnvironment.getMessager().printMessage(Diagnostic.Kind.ERROR, this.getMessage(), this.element);
    }
}

File: http/http-server-symbol-processor/src/main/kotlin/ru/tinkoff/kora/http/server/symbol/procesor/exception/HttpProcessorException.kt
package ru.tinkoff.kora.http.server.symbol.procesor.exception

import com.google.devtools.ksp.processing.KSPLogger
import com.google.devtools.ksp.symbol.KSNode

class HttpProcessorException(message: String, private val ksNode: KSNode) : RuntimeException(message, null) {
    fun printError(logger: KSPLogger) {
        logger.error(message!!, ksNode)
    }
}
